src/main/java/ratpack/pac4j/internal/RatpackWebContext.java

```diff
@@ -65,22 +79,29 @@ public RatpackWebContext(Context ctx, TypedData body, SessionData session) {
     Promise<SessionData> sessionDataPromise = ctx.get(Session.class).getData();
     if (bodyBacked) {
       return ctx.getRequest().getBody().flatMap(body ->
-        sessionDataPromise.map(sessionData -> new RatpackWebContext(ctx, body, sessionData))
+          sessionDataPromise.map(sessionData -> new RatpackWebContext(ctx, body, sessionData))
       );
     } else {
       return sessionDataPromise.map(sessionData -> new RatpackWebContext(ctx, null, sessionData));
     }
   }
 
-  @Override
   public SessionStore getSessionStore() {
     return this.session;
   }
 
+  public ProfileManager getProfileManager() {
+    return profileManager;
+  }
+
   @Override
-  public String getRequestParameter(String name) {
-    return Optional.ofNullable(request.getQueryParams().get(name))
-      .orElseGet(() -> form.get(name));
+  public Optional<String> getRequestParameter(String name) {
+    Optional<String> param = Optional.ofNullable(request.getQueryParams().get(name));
+    if(param.isPresent()) {
```

Comment:
`orElseGet` or some other variant should be able to replace this if statement.

Reply:
The `else` result is `Optional` too, so I would need to use `or`, but that's in 9 and we're targeting 1.8.